pcdsutils/ophyd_helpers.py

```diff
@@ -0,0 +1,242 @@
+"""
+ophyd Device/Signal-related helpers.
+"""
+
+import asyncio
+import contextlib
+import logging
+import time
+from typing import Callable, List, Optional
+
+import ophyd
```

Comment:
New dependency? Do we add this to the meta.yaml/requirements.txt or do we take it as optional? If optional, is it ok to leave it exposed and assume nobody imports here if they don't have ophyd?

Previously we had an implied ophyd dependency in the log submodule, but it was not tied to an import.

Reply:
My feeling is to keep it optional, I think. To recap/add some thoughts:
* It's currently in `dev-requirements.txt` for the existing test suite.
* I could make it fail more gracefully, perhaps, if we think it importing ophyd_helpers without ophyd could ever be a potential issue.
* Maybe the dependency tree would illuminate how big a deal additional deps would be for pcdsutils. Would you by chance have a tree handy of all our packages that depend on pcdsutils currently?

Reply:
I'm happy to leave it like this, but it felt like an important thing to at least exchange words about.
```
$ mamba repoquery whoneeds pcdsutils

 Name         Version Build        Depends           Channel
─────────────────────────────────────────────────────────────
 pcdsdevices  5.1.0   pyhd8ed1ab_0 pcdsutils >=0.4.0
 typhos       2.2.1   pyhd8ed1ab_0 pcdsutils
 lucid        0.9.0   pyhd8ed1ab_0 pcdsutils
 pmgr         2.0.2   pyhd8ed1ab_1 pcdsutils
 hutch-python 1.13.2  py_1         pcdsutils >=0.6.0
```

Reply:
Of the above set, only `pmgr` doesn't have `ophyd` as a dependency.

In general, as long as this doesn't become a top-level import I see no problem in leaving this as an optional dependency with a loud import failure.

Reply:
> I'm happy to leave it like this, but it felt like an important thing to at least exchange words about.

This is 100% something that should be brought up in the review - thank you.

> In general, as long as this doesn't become a top-level import I see no problem in leaving this as an optional dependency with a loud import failure.

I think I agree. For a utility library like this, I think I like the pattern of "use what you like, just make sure you have the deps in your project for what you use".

Reply:
Ok, great. If ever there comes a day where we feel very strongly about a non-ophyd utility existing in ophyd_helpers we can revisit (let's not do that).